onadata/apps/logger/signals.py

```diff
@@ -1,9 +1,44 @@
 # coding: utf-8
 import logging
-from django.db.models.signals import pre_delete
+
+from django.db.models import F
+from django.db.models.signals import (
+    pre_delete,
+    post_save,
+)
 from django.dispatch import receiver
 
 from onadata.apps.logger.models.attachment import Attachment
+from onadata.apps.logger.models.xform import XForm
+from onadata.apps.main.models.user_profile import UserProfile
+
+
+@receiver(pre_delete, sender=Attachment)
+def delete_user_profile_attachment_storage_bytes(instance, **kwargs):
+    """
+    Update the attachment_storage_bytes field in the UserProfile model
+    when an attachment is deleted
+    """
+    file_size = instance.media_file.size
+    owner_profile = instance.instance.xform.user.profile.pk
+    queryset = UserProfile.objects.filter(pk=owner_profile)
+    queryset.update(
+        attachment_storage_bytes=F('attachment_storage_bytes') - file_size
+    )
+
+
+@receiver(pre_delete, sender=Attachment)
+def delete_xform_attachment_storage_bytes(instance, **kwargs):
+    """
+    Update the attachment_storage_bytes field in the XForm
+    model when an attachment is deleted
+    """
+    file_size = instance.media_file.size
+    xform = instance.instance.xform.pk
+    queryset = XForm.objects.filter(pk=xform)
```

Comment:
my same comment from `delete_user_profile_attachment_storage_bytes` about `media_file.size` applies here as well.

for query efficiency, i think the only improvement to be made is to change `instance.instance.xform.pk` to `instance.instance.xform_id`. the former actually loads the entire `XForm` object into memory, while the latter takes advantage of the `Instance` object actually having the ID of the related `XForm` in memory already.